### PR TITLE
Fix incorrect lodash to javascript conversions

### PR DESCRIPTION
**Pull request by icirellik**
* Correctly detect window
* Fix lodash conversions (really need tests, next PR)
* Update children PropTypes

### CODE REVIEW

File: components/Carousel.jsx [hunk 1]
```diff
@@ -38,13 +46,9 @@ var Carousel = React.createClass({
   },
 
   getCurrentOffset: function () {
-    var currentIndexOffset = this.state.itemStart || this.state.currentIndex;
-
-    if (currentIndexOffset) {
-      return 0;
-    }
+    var currentIndexOffset = this.state.itemStart[this.state.currentIndex];
```
Comment (icirellik) [inline components/Carousel.jsx]: Original line: 

```js
var currentIndexOffset = _.get(this.state.itemStart, this.state.currentIndex);
```

File: components/Carousel.jsx [hunk 2]
```diff
@@ -30,7 +33,12 @@ var Carousel = React.createClass({
 
     // all the same size
     var cardCount = carouselContainer.children.length;
-    var cardWidth = carouselContainer.children[0].offsetWidth;
+    var cardWidth;
+    if (React.Children.count === 1) {
```
Comment (pwmckenna) [inline components/Carousel.jsx]: iirc count is a function that takes the children prop....
```
React.Children.count(carouselContainer.children)
```

File: components/Carousel.jsx [hunk 3]
```diff
@@ -10,7 +10,10 @@ React.initializeTouchEvents(true);
 var Carousel = React.createClass({
 
   propTypes: {
-    children: React.PropTypes.object
+    children: React.PropTypes.oneOfType([
+      React.PropTypes.array,
```
Comment (pwmckenna) [inline components/Carousel.jsx]: array/object of what? React.PropTypes.element?

**Comment by icirellik** [comment]
Updated to follow the eslint-plugin-react documentation https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md

**Pull request merged**